In [2]:
# import the libraries needed

import pandas as pd
import numpy as np
import os
import sys
sys.path.append(os.path.abspath(os.path.join('../scripts')))
import seaborn as sns
import matplotlib.pyplot as plt
import plotly.express as px



In [4]:
# ignore the warnings
import warnings
warnings.filterwarnings('ignore')

In [6]:
#load the data that we are to use
df = pd.read_csv("C:/Users/DESMOND/Airflight_analysis/AirportMapping.csv")
df.head()

City  Country Code
0         Aalborg  Denmark  AAL
1        Aalesund   Norway  AES
2          Aarhus  Denmark  AAR
3  Abbotsford, BC   Canada  YXX
4  Abbotsford, BC   Canada  YXX

### EDA

In [7]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1579 entries, 0 to 1578
Data columns (total 3 columns):
 #   Column   Non-Null Count  Dtype 
---  ------   --------------  ----- 
 0   City     1579 non-null   object
 1   Country  1578 non-null   object
 2   Code     1578 non-null   object
dtypes: object(3)
memory usage: 37.1+ KB


the data we have has got 1579  cities, 1578 countries and codes. 

In [10]:
class OverViewAnalysis:
    
    def __init__(self, df):
        
        self.df = df
    
    
    def read_head(self, top=5):
        return self.df.head(top)
    
    # returning the number of rows columns and column information
    def get_info(self):
        row_count, col_count = self.df.shape
    
        print(f"Number of rows: {row_count}")
        print(f"Number of columns: {col_count}")
        print("================================")

        return (row_count, col_count), self.df.info()
    
    # gets number of distnict values in a given coumn
    def get_count(self, column_name):
        return self.df[column_name].value_counts()
    
    # getting the null count for every column
    def get_null_count(self, column_name):
        print("Null values count")
        print(self.df.isnull().sum())
        return self.df.isnull().sum()
    
    
    

In [11]:
overViewAnal = OverViewAnalysis(df)

In [12]:
overViewAnal.read_head()

City  Country Code
0         Aalborg  Denmark  AAL
1        Aalesund   Norway  AES
2          Aarhus  Denmark  AAR
3  Abbotsford, BC   Canada  YXX
4  Abbotsford, BC   Canada  YXX

In [13]:
overViewAnal.get_info()

Number of rows: 1579
Number of columns: 3
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1579 entries, 0 to 1578
Data columns (total 3 columns):
 #   Column   Non-Null Count  Dtype 
---  ------   --------------  ----- 
 0   City     1579 non-null   object
 1   Country  1578 non-null   object
 2   Code     1578 non-null   object
dtypes: object(3)
memory usage: 37.1+ KB


((1579, 3), None)

Now we have the overview analysis of the data that we are working with. we can see the missing values and we deduce there are none
the data types that we have. 

In [16]:
df.isnull().sum()

City       0
Country    1
Code       1
dtype: int64

We have one missinf data in the code and also in the country